android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -372,7 +372,10 @@ class GeckoEngine(
             }
 
             override fun onInstalled(extension: org.mozilla.geckoview.WebExtension) {
-                webExtensionDelegate.onInstalled(GeckoWebExtension(extension, runtime))
+                val installedExtension = GeckoWebExtension(extension, runtime)
+                webExtensionDelegate.onInstalled(installedExtension)
+                installedExtension.registerActionHandler(webExtensionActionHandler)
+                installedExtension.registerTabHandler(webExtensionTabHandler, defaultSettings)
```

Comment:
The patch is working but I was wondering why it's not working with [webExtensionDelegate.onInstalled](https://github.com/mozilla-mobile/firefox-android/blob/3b39827f151c5d7ee7ff44792b45b78ef005235e/android-components/components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt#L230C1-L230C1) as it looks like is doing the same, ideally we would like to have that logic in shared point.

Reply:
Same with [updateWebExtension](https://github.com/mozilla-mobile/firefox-android/blob/3b39827f151c5d7ee7ff44792b45b78ef005235e/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#L299-L300)

Reply:
As we are using it when installing from [GeckoEngine](https://github.com/mozilla-mobile/firefox-android/blob/3b39827f151c5d7ee7ff44792b45b78ef005235e/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#L232-L233) which is also calling `webExtensionDelegate.onInstalled`, we need it here too.